verify/test-repo-infra-subtree.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+set -o pipefail
+
+readonly WORKSPACE="$( mktemp -d )"
+trap 'rm -rf -- "$WORKSPACE"' EXIT
+
+[ "${TRAVIS:-}" = 'true' ] && {
+  git checkout "$BRANCH_TO_MERGE"
+}
+
+readonly REPO_INFRA_ROOT="$( git rev-parse --show-toplevel )"
```

Comment:
if this is running in another repo where it's been subtree'd, is this the root of that repo, or the root of that subtree?

Reply:
Note that this is "just" the test for the `{update,verify}-repo-infra-subtree.sh` scripts, it is supposed to run locally (while maybe hacking on `{update,verify}-repo-infra-subtree.sh` **and** in CI

The flow of the test is:
  1. create a temporary repo
  2. `subtree add` the current version (the PR branch) into the temporary repo
  3. run `{update,verify}-repo-infra-subtree.sh` in the temporary repo and check if they do what they promised

`$REPO_INFRA_ROOT` is supposed to be the local path to the git repo of the version of `k/repo-infra` that is under test / to be merged. `$REPO_INFRA_ROOT` is the path that will be used when the test [`subtree add`s](https://github.com/kubernetes/repo-infra/pull/94/files/b7b803d1efca6335e42311935df0ed2fe7cb9853#diff-06847f1dec8f17f07a568abdbcef5e06R71) `repo-infra` into the temporary repo.

I notice now that this test assumes to be run within the `repo-infra` directory. This just works fin in CI, but might break locally. I can see if I can guard against that for local invocations.

Note:
There was the request in [another review](https://github.com/kubernetes/repo-infra/pull/94#pullrequestreview-191674251) to rename that variable from `REPO_INFRA_URL` to `REPO_INFRA_ROOT` which might make more sense in the context of the test. Or maybe it doesn't. I am bad at naming.